frontend/src/app/pages/user/user.component.ts

```diff
@@ -0,0 +1,113 @@
+import { Component } from '@angular/core';
+import { OnDestroy, OnInit } from '@angular/core';
+import { UserService } from 'src/app/core/services/user.service';
+import { UserDTO } from 'src/app/core/models/user-dto';
+import { UpdateUserDTO } from 'src/app/core/models/update-user-dto';
+import { ActivatedRoute, Router } from '@angular/router';
+import { AbstractControl, FormBuilder, FormControl, FormGroup, ValidationErrors, ValidatorFn, Validators } from '@angular/forms';
+import { finalize, Observable, of, Subscription, switchMap, tap } from 'rxjs';
+import { MessageService } from 'primeng/api';
+import { HttpClient } from '@angular/common/http';
+
+@Component({
+  selector: 'app-user',
+  templateUrl: './user.component.html',
+  styleUrls: ['./user.component.scss']
+})
+export class UserComponent implements OnInit, OnDestroy {
+  id!: string | null;
+  user!: UserDTO;
+  userForm!: FormGroup;
+  subscriptions: Subscription[] = [];
+  isProgressSpinnerVisible = false;
+
+  constructor(
+    private route: ActivatedRoute, 
+    private userService: UserService,
+    private router: Router,
+    private formBuilder: FormBuilder,
+    private messageService: MessageService,
+  ) {
+    this.id = null;
+    this.userService.getUser(this.id).pipe(tap(res => this.redo(res))).subscribe(user => this.user = user)!;
+  }
+
+  private redo(userDTO: UserDTO):void {
+    this.id = userDTO.id;
+    this.userForm.get('name')?.setValue(userDTO.name);
+    this.userForm.get('surname')?.setValue(userDTO.surname);
+    this.userForm.get('nickname')?.setValue(userDTO.nickname);
+    this.userForm.get('userType')?.setValue(userDTO.userType == 'Simple' ? false : true);
+    this.userForm.get('email')?.setValue(userDTO.email);
+    this.userForm.get('accountBallance')?.setValue(userDTO.accountBalance);
+  }
+
+  ngOnInit(): void {
+    this.userForm = this.formBuilder.group({
+      name: new FormControl('', Validators.required),
+      surname: new FormControl('', Validators.required),
+      nickname: new FormControl('', Validators.required),
+      // avatar TODO
```

Comment:
// avatar TODO?

Reply:
it is because of the issue with downloading image from azure